### PR TITLE
HHH-17671 Test and fix for SqmPredicate copy failure

### PR DESCRIPTION
**Pull request by jrenaat**
https://hibernate.atlassian.net/browse/HHH-17671

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmBasicValuedSimplePath.java [hunk 1]
```diff
@@ -45,12 +45,13 @@ public SqmBasicValuedSimplePath(
 
 	@Override
 	public SqmBasicValuedSimplePath<T> copy(SqmCopyContext context) {
+		final SqmPath<?> lhsCopy = getLhs().copy( context );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmBasicValuedSimplePath.java:48]: Please revert the changes here and instead remove the copying of the `reusablePaths` in `org.hibernate.query.sqm.tree.domain.AbstractSqmPath#copyTo`.

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmBasicValuedSimplePath.java:48]: Still, please revert the change in this class.

Comment (jrenaat) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmBasicValuedSimplePath.java]: Sorry, forgot that.
